### PR TITLE
st1 [2188][ADD] sale_order_country

### PR DESCRIPTION
**Pull request by tack-yabeQuartile**
[2188](https://www.quartile.co/web#id=2188&action=771&model=project.task&view_type=form&menu_id=505)

* Display country of partner, filter japan or the others, group by countries in list views of sale orders and quotations.

### CODE REVIEW

**Comment by tack-yabeQuartile** [comment]
I'll check pre-commit later.

File: sale_order_display_country/views/sale_order_views.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                    string="Oversea Customers"
+                    name="oversea"
+                    domain="[('country_code', '!=', 'JP')]"
+                />
+            </xpath>
+            <xpath expr="//filter[@name='customer']" position="after">
+                <filter
+                    string="countries"
+                    name="by_country"
+                    domain="[]"
```
Comment (yostashiro) [inline sale_order_display_country/views/sale_order_views.xml]:
```suggestion
```

File: sale_order_display_country/views/sale_order_views.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        <field name="arch" type="xml">
+            <xpath expr="//filter[@name='my_sale_orders_filter']" position="before">
+                <filter
+                    string="Japanese Customers"
+                    name="jp"
+                    domain="[('country_code', '=', 'JP')]"
+                />
+                <filter
+                    string="Oversea Customers"
+                    name="oversea"
```
Comment (yostashiro) [inline sale_order_display_country/views/sale_order_views.xml]:
```suggestion
                    string="Overseas Customers"
                    name="overseas"
```

File: sale_order_display_country/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+from odoo import fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    country = fields.Many2one(
+        related="partner_id.country_id", readonly=True, store=True,
+    )
```
Comment (yostashiro) [inline sale_order_display_country/models/sale_order.py]:
```suggestion
    country_id = fields.Many2one(related="partner_id.country_id", store=True)
```
Should be readonly=True by default for compute/related fields.

Many2one field should be defined as `xxx_id`.  See https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#fields

File: sale_order_display_country/models/sale_order.py [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    country_code = fields.Char(related="partner_id.country_id.code")
```
Comment (yostashiro) [inline sale_order_display_country/models/sale_order.py]:
```suggestion
    country_code = fields.Char(related="partner_id.country_id.code", store=True)
```
May be worth it to store the value for performance reason if the filter is to be used frequently.

**Review by yostashiro — CHANGES_REQUESTED** [review]
Can we perhaps change the module name to sale_order_country.

Is there no similar requirement on the invoice?

**Comment by tack-yabeQuartile** [comment]
Module name has been changed.

**Review by yostashiro — APPROVED** [review]
LGTM 👍

**Pull request merged**